### ISSUE TITLE
setStyle (dynamic styles without resending entire Vector layers)

### ISSUE THREAD

**Issue by cholcomb** [rstudio/leaflet#152]
It would be awesome to have some way to interactively change the styles of individual features of `GeoJSON` objects (and possibly other vector layers) without having to remove and re-add the entire layer. FWIW it appears this works for `TopoJSON` as well: http://gis.stackexchange.com/questions/120012/leaflet-omnivore-style-function. The final answer on this question (http://gis.stackexchange.com/questions/75590/setstyle-function-for-geojson-features-leaflet) included this:

```Javascript
geojson_layer.eachLayer(function (layer) {  
  if(layer.feature.properties.NAME == 'feature 1') {    
    layer.setStyle({fillColor :'blue'}) 
  }
});
```

As this is my first exposure to this method, I don't have detailed recommendations on how to implement this yet.   I can describe my use case - I'm dynamically calculating some values in R that are used to color `GeoJSON` features, so right now I'm removing the `GeoJSON` and adding back a `GeoJSON` with the new styles upon each calculation. BTW, the `jsonlite` and `RJSONIO` `toJSON` functions are very slow for converting back to `GeoJSON` after manipulating the data as an `R` list (takes around 15 seconds for a 1MB `GeoJSON` with only id's and simplified shapes), so I am actually parsing the `GeoJSON` as text and inserting the styles (0.3 seconds).

**Comment by cholcomb** [comment]
A example that shows `setStyle` as well as `resetStyle`) is at http://leafletjs.com/examples/choropleth.html. This also shows the excellent speed improvements of mouseover, mouseout, and click that can occur if that's done without `R` intercepting (is there a way to input arbitrary JS functions for these events for each layer?). Note: the ability to color the borders quickly is critical because with colored polygons may hide the border (especially for multi-polygons).

I edited the prior example to include the aforementioned code to turn Texas blue (it changes back after a mouseout) into this jsfiddle (removed some code from original example): http://jsfiddle.net/7xqt2crh/2/.

**Comment by timelyportfolio** [comment]
:+1:

**Comment by jcheng5** [comment]
Really nice idea...

**Comment by mbacou** [comment]
Yes, ran into that use case yesterday. Restyling/re-using layers is very common scenario.

**Comment by cholcomb** [comment]
FWIW, I made a branch with a simplified version of some of these
capabilities and included a new example utilizing them. I'm on vacation now
but may clean them up further and send over a PR once I have time.

**Comment by ComeMaes** [comment]
Hi @ideamotor,

Have you worked any further on this? I'd be very interested in such a feature and willing to contribute, as I also feel that it might help in a variety of fairly common scenarios.

**Comment by timelyportfolio** [comment]
#496
